addons/udes_common/models/models.py

```diff
@@ -84,3 +85,23 @@ def statistics(self, cache=False):
 def trace(self, filter=None, max=None):
     """Trace database queries"""
     return tools.QueryTracer(self.env.cr, filter=filter, max=max)
+
+
+@add_if_not_exists(models.BaseModel)
+def selection_value_displayname(self, selection_field_name):
+    """Get the display name - for a given selection fields value - on a recordset
+    :param: self: <some.model> recordset (len 1) to get the value from
+    :param: selection_field_name: str() field name to get the value for
+
+    :returns: str() Display name of the currently set selection field value of the recordset
+    """
+    self.ensure_one()
+    try:
+        selection_field = self._fields[selection_field_name]
+    except KeyError:
+        raise ValidationError(
+            _(f"Field '{selection_field_name}' does not exist on model '{self._name}'")
+        )
+    # Odoo has a nice function that handles all the weird ways you can define selections!
+    selection_values_dict = dict(selection_field._description_selection(self.env))
+    return selection_values_dict.get(self[selection_field_name], False)
```

Comment:
Not sure if I'm misunderstanding something but if we can't find a display name for the value I think it is better to return the code value rather than `False`.

Reply:
This is more so if the selection fields value is set to False, the .get() will return None.
Just trying to keep consistency with how Odoo behaves when getting the normal selection field value